packages/@aws-cdk/custom-resources/lib/aws-custom-resource/runtime/index.ts

```diff
@@ -115,10 +115,42 @@ export async function handler(event: AWSLambda.CloudFormationCustomResourceEvent
     const call: AwsSdkCall | undefined = event.ResourceProperties[event.RequestType];
 
     if (call) {
-      const awsService = new (AWS as any)[call.service]({
-        apiVersion: call.apiVersion,
-        region: call.region,
-      });
+      let awsService;
+
+      if (call.assumedRoleArn) {
+
+        const stsService = new (AWS as any).STS({
+          apiVersion: call.apiVersion,
```

Comment:
The API version doesn't go here.

Reply:
But instead of this manual call, I'd prefer you use `ChainableTemporaryCredentials`.

https://docs.aws.amazon.com/ko_kr/AWSJavaScriptSDK/latest/AWS/ChainableTemporaryCredentials.html

See here for example usage:

https://github.com/aws/aws-cdk/blob/master/packages/aws-cdk/lib/api/aws-auth/sdk-provider.ts/#L314

Reply:
Magic, I like it! Changed to using it now